src/odb/src/cdl/cdl.cpp

```diff
@@ -105,7 +106,14 @@ std::string getNetName(dbBlock* block,
     return getUnconnectedNet(block, unconnectedNets);
   }
 
-  return net->getName();
+  std::string net_name = net->getName();
+
+  auto pin_name_match = net2pin.find(net_name);
+  if (pin_name_match != net2pin.end()) {
+    return pin_name_match->second;
+  }
```

Comment:
Rather than build net2pin and lookup, you could just handle the bterm here, eg
```
dbBTerm* bterm = net->get1stBTerm();
if (bterm) {
  return bterm->getName();
}
return net->getName();
```